proto/Handshake.proto

```diff
@@ -4,11 +4,9 @@ package proto;
 
 import "UiOptions.proto";
 
+// A message which contains either a list of declarations or a list of values for a single module.
 message Handshake {
-    string name = 1;
-    repeated UiOption options = 2;
-}
-
-message HandshakeAccumulation {
-    repeated Handshake handshakes = 1;
+    string module_name = 1;
+    UiOptionDeclarations declarations = 2;
```

Comment:
Could we make this optional? I believe we would only send this at the beginning or when the AI wishes to make changes to the declaration of a field to make it (im)mutable

Reply:
We could make the two fields a oneof though, ensuring that only one of the two is sent.

Reply:
Though I don't personally think that makes a lot of sense here.

Reply:
Yeah, it doesn't make sense to use oneof. I was not aware that in proto3 singular messages have explicit presence tracking by default.